### PR TITLE
added VIA support for HotDox

### PR DESCRIPTION
**Pull request by unicorn855**
## Description

I adjusted the VENDOR_ID and PRODUCT_ID in the config.h file of the keyboard's base directory such that they are unique. Further, the config.h in the /via subdirectory contains instructions of the maximum layer count VIA requires. Lastly, the keymap.c in the /via subdirectory is an adjusted copy of the default keymap.c with the three additional layers specifically assigned KC_TRANS codes so as to avoid them being populated randomly when not used. I did not delete any of the additional code or inline documentation present in the default keymap, so that once mouse layers and media keys are supported by VIA, those can easily be found and re-added to this keymap. Rules.mk enable VIA as well as disable currently unsupported features, mitigating errors due to those yet unsupported features.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap (addition)
- [ ] Documentation

## Issues Fixed or Closed by This PR

 (not applicable)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage). read: 'make hotdox:via' compiles without errors.

### CODE REVIEW

File: keyboards/hotdox/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+// Runs just one time when the keyboard initializes.
+void matrix_init_user(void) {
+
+};
+
+
+// Runs constantly in the background, in a loop.
+void matrix_scan_user(void) {
+
+    uint8_t layer = biton32(layer_state);
```
Comment (drashna) [inline keyboards/hotdox/keymaps/via/keymap.c:193]:
```suggestion
    uint8_t layer = get_highest_layer(layer_state);
```

**Comment by unicorn855** [comment]
Is there something I still should be looking into? Thank you for the suggestion @drashna :)

**Comment by unicorn855** [comment]
I am confused as to why @klackygears' merge pull request has thrown an error upon (presumably automatic) testing? Is there something I can do to help, or has this issue been resolved already?

**Comment by Erovia** [comment]
@unicorn855 some of the tests for the CLI were broken. A fix has been merged now.

**Comment by unicorn855** [comment]
@Erovia ah I see, thank you for clearing that up :)

**Pull request merged**